CorsixTH/Src/th_gfx_sdl.cpp

```diff
@@ -95,6 +97,57 @@ uint8_t convert_6bit_to_8bit_colour_component(uint8_t colour_component) {
       (colour_component & mask_6bit) * static_cast<double>(0xFF) / mask_6bit));
 }
 
+double hueToRGB(double p, double q, double hue) {
+  if (hue < 0) {
```

Comment:
Should these be `while` instead of if - or is hue only ever slightly out of range from the 1/3rd adjustment?